### DISCUSSION TITLE
Get topologically equivalent atoms

### DISCUSSION THREAD

**Discussion by TheophileGaudin** [rdkit/rdkit#5808] [Q&A] [answered]
Hi,

I wonder whether it is possible to get a list of topologically equivalent atoms with rdkit?

For example the SMILES of neopentane is CC(C)(C)C. We have here 5 C atoms, with indices 0, 1, 2, 3, 4. All 4 external C atoms are topologically equivalent though, so canonical ordering of the C atoms won't really distinguish between them. How to have for each atom a list of indices of topologically equivalent atoms? Here let's say the central atom is index 0. We would have as desired result:

```
[[],[2,3,4],[1,3,4],[1,2,4],[1,2,3]]
```

Thanks in advance!

**Comment by bp-kelley** [comment, marked as answer]
see CanonicalRankAtoms, with breakTies=False, this gives the symmetry order for each atom, equivalent symmetries are identical topographically.

```
>>> m = Chem.MolFromSmiles("CC(C)(C)C")
>>> list(Chem.CanonicalRankAtoms(m, breakTies=False))
[0, 4, 0, 0, 0]
```
